### PR TITLE
PRO-2344 & PRO-3236: Save CCD case before payment and update status after payment

### PR DESCRIPTION
**Pull request by dhruvpatel20**
This pull request contains:

PRO-2344 - saving the CCD case via the submit endpoint.
PRO-3236 - updating the CCD case via a new endpoint "updatePaymentStatus".
Added unit tests, updated existing tests and component tests.

Added "nock" as a dev dependency for the tests, which allows mocking of end points - its industry standard and other teams in HMCTS use it. Its basically a more effective way than having the stub services.

@reddy1praveen I have had to make changes to the terraform variables as we now use 2 endpoints in the submit service.

This PR relies on: https://github.com/hmcts/probate-submit-service/pull/65

### CODE REVIEW

File: test/unit/testSubmitService.js [hunk 1]
```diff
@@ -1,60 +1,63 @@
...
+
+  afterEach(function () {
+    nock.cleanAll();
+  });
+
+  it('Should successfully update payment status', function (done) {
+    const expectedResponse = {};
+    nock(SUBMIT_SERVICE_URL).post('/updatePaymentStatus')
+      .reply(200, expectedResponse);
+    services.updateCcdCasePaymentStatus(formData, ctx);
```
Comment (shepda) [inline test/unit/testSubmitService.js]: I think we should check that the return value of this method is what we're expecting after a successful post

File: test/unit/testSubmitService.js [hunk 2]
```diff
@@ -1,60 +1,63 @@
...
+    done();
+  });
+
+  it('Should successfully sent to submit service', function (done) {
+    const expectedResponse = {};
+    const scope = nock(SUBMIT_SERVICE_URL).post('/submit')
+      .reply(200, expectedResponse);
+    services.sendToSubmitService(formData, ctx);
```
Comment (shepda) [inline test/unit/testSubmitService.js]: I think we should check that the return value of this method is what we're expecting after a successful post

File: test/unit/testSubmitService.js [hunk 3]
```diff
@@ -1,60 +1,63 @@
...
+  it('Should successfully find payment', function (done) {
+    nock(`${CREATE_PAYMENT_SERVICE_URL.replace('userId', 123)}`).get('/1')
+      .reply(200, {
+      'state': {
+        'status': 'success'
+      }
     });
+    formData.paymentId = 1;
+    formData.userId = 123;
+    services.findPayment(formData);
```
Comment (shepda) [inline test/unit/testSubmitService.js]: I think we should check that the return value of this method is what we're expecting after a successful post

File: test/unit/testSubmitService.js [hunk 4]
```diff
@@ -1,60 +1,63 @@
...
+    done();
+  });
 });
```
Comment (shepda) [inline test/unit/testSubmitService.js]: Some lines in this file are indented with 2 spaces, indentation should be 4 spaces.

File: app/config.js [hunk 5]
```diff
@@ -25,9 +25,8 @@ module.exports = {
             url: process.env.VALIDATION_SERVICE_URL || 'http://localhost:8080/validate'
         },
         submit: {
-            url: process.env.SUBMIT_SERVICE_URL || 'http://localhost:8181/submit',
-            port: 8181,
-            path: '/submit'
+            url: process.env.SUBMIT_SERVICE_URL || 'http://localhost:8181',
+            port: 8181
```
[the remark this answers is not in the document]
Comment (dhruvpatel20) [reply app/config.js]: I thought it better just to keep the URL here, as we are using 2 different endpoints for the submit service - and only referenced in services.json

File: app/steps/ui/payment/status/index.js [hunk 6]
```diff
@@ -67,48 +67,39 @@ module.exports = class PaymentStatus extends Step {
...
+      return options;
     }
 
-    * sendApplication(ctx, formdata) {
+    * updateCcdCasePaymentStatus(ctx, formdata) {
         const submitData = {};
-        const softStop = this.anySoftStops(formdata, ctx) ? 'softStop' : false;
         Object.assign(submitData, formdata);
-
-        const result = yield services.submitApplication(submitData, ctx, softStop);
```
[the remark this answers is not in the document]
Comment (dhruvpatel20) [reply app/steps/ui/payment/status/index.js]: The backend is expecting the data as submit data, I carried on the same pattern with the new endpoint. I also persist the submit data on the backend.

File: app/config.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (shepda) [reply app/config.js]: Ok...the path is also used in `test/service-stubs/submit.js` but there's another comment about changing this to use the url

File: app/steps/ui/payment/status/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (shepda) [reply app/steps/ui/payment/status/index.js]: So it's only a naming convention then? As the object for `submitData` and `formdata` would be the same.

File: app/steps/ui/payment/status/index.js [hunk 7]
```diff
@@ -66,11 +66,12 @@ module.exports = class PaymentStatus extends Step {
             };
 
             const findPaymentResponse = yield services.findPayment(data);
+            const date = findPaymentResponse.date_updated == undefined ? ctx.paymentCreatedDate : findPaymentResponse.date_updated;
```
[the remark this answers is not in the document]
Comment (dhruvpatel20) [reply app/steps/ui/payment/status/index.js]: 👍

File: app/steps/ui/payment/status/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (dhruvpatel20) [reply app/steps/ui/payment/status/index.js]: had a convo with you, understand now!

File: test/service-stubs/submit.js [hunk 8]
```diff
@@ -9,7 +9,7 @@ const router = require('express').Router();
 const SUBMIT_SERVICE_PORT = config.services.submit.port;
 const SUBMIT_SERVICE_PATH = config.services.submit.path;
```
[the remark this answers is not in the document]
Comment (dhruvpatel20) [reply test/service-stubs/submit.js]: 👍

**Pull request closed without merging**